88 mvp 0   basic login and user data persistence

Pharo-14 opened the pull request:
Server.js is created and connected to mongoDB

Commit:
new json files for dependencies

Commit:
implemented location tracking with user permission

Commit:
Merge branch '26-task-111-integrate-map-service-and-setup-map-interface' into 37-task-141-implement-functionality-to-identify-users-current-location

Commit:
implemented the required functionality with task's #111 map and added the location tracking permission prompt

mE3lissa committed:
Refactor BuildingRepository and campus repository to handle connection internally

vibqetowi committed:
Merge branch 'main' of https://github.com/AsifAliKhan2001/Minicap into 37-task-141-implement-functionality-to-identify-users-current-location

Commit:
Added a refresh user location button and implemented live tracking

Commit:
Added a refresh user location button and implemented live tracking

Amine-Arrachid committed:
login frontend

Login UI (frontend)

Commit:
Added a refresh user location button and implemented live tracking

Younesbhb committed:
Update Login.tsx

merged the email and student id, that way the user can log in by either entering their student id or email

Pharo-14 committed:
Created server.js and configured MongoDB connection

mE3lissa committed:
modified buildingViewModel, buildingRepository and building model to add a method which can fetch a building from the database given its UUID

mE3lissa committed:
fixed indentation

mE3lissa committed:
Merge branch 'main' into 107--mvp0---Database-Setup---Campus

vibqetowi reviewed (changes requested):
Refactoring required, this needs to follow the user model defined in MVVM, also please implement more repo methiods than this. Sign up login

refactoring to gluestack deferred because UI team still hasnt opened PR

vibqetowi committed:
fixed issues with manual refresh

vibqetowi committed:
rename update locaton button

vibqetowi committed:
Merge branch '37-task-141-implement-functionality-to-identify-users-current-location' of https://github.com/AsifAliKhan2001/Minicap into 107--mvp0---Database-Setup---Campus

vibqetowi committed:
Merge branch '37-task-141-implement-functionality-to-identify-users-current-location' of https://github.com/AsifAliKhan2001/Minicap into 107--mvp0---Database-Setup---Campus

vibqetowi committed:
save because im getting lost

vibqetowi committed:
added authto  interfaces

vibqetowi committed:
change to mongo object id instead of UUID

vibqetowi committed:
refactoring done

vibqetowi committed:
delete artefact

vibqetowi committed:
using @imports

vibqetowi committed:
Merge branch '107--mvp0---Database-Setup---Campus' of https://github.com/AsifAliKhan2001/Minicap into 88-mvp-0---basic-login-and-user-data-persistence

vibqetowi committed:
Merge branch '107--mvp0---Database-Setup---Campus' of https://github.com/AsifAliKhan2001/Minicap into 88-mvp-0---basic-login-and-user-data-persistence

Amine-Arrachid committed:
Method implementation

Added findUserById, signUp, login, deleteUser, and findByEmail as required by the UserRepository. I also implemented a private helper method to generate token. I added error messages for invalid inputs as well.

Amine-Arrachid committed:
update user details

Implemented the update user details models

Amine-Arrachid committed:
Update uvvm

Implemented logout method

Amine-Arrachid committed:
Update UserViewModel.ts

UpdateAudit comment waiting to be implemented

Pharo-14 committed:
fixed package.json

Pharo-14 committed:
Merge branch '88-mvp-0---basic-login-and-user-data-persistence' of https://github.com/AsifAliKhan2001/Minicap into 88-mvp-0---basic-login-and-user-data-persistence
